### ISSUE TITLE
Question About 

### ISSUE THREAD

**Issue by ontocord** [UKPLab/sentence-transformers#373]
Hi, 

Wonderful library you have written!

In MultipleNegativesRankingLoss.py, you have this code:

        scores = torch.matmul(embeddings_a, embeddings_b.t())
        diagonal_mean = torch.mean(torch.diag(scores))
        mean_log_row_sum_exp = torch.mean(torch.logsumexp(scores, dim=1))
        return -diagonal_mean + mean_log_row_sum_exp

Shouldn't you take into account the diagonal so that it isn't in the mean_log_row_sum_exp value? 

Thank you!

**Comment by nreimers** [comment]
Hi @ontocord 
No, the math is correct. It is a more efficient computation of computing negative log-likelihood with softmax for a batch with the given properties.

Best
Nils Reimers

**Comment by ontocord** [comment]
Thanks Nil for the answer!
